langchain4j-agentic-a2a/src/main/java/dev/langchain4j/agentic/a2a/DefaultA2AClientBuilder.java

```diff
@@ -109,23 +125,46 @@ private Object invokeAgent(Object[] args) throws A2AServerException {
             }
         }
 
-        Message message = new Message.Builder()
-                .role(Message.Role.USER)
-                .parts(parts)
-                .build();
-
-        MessageSendParams params = new MessageSendParams.Builder()
-                .message(message)
-                .build();
-
-        SendMessageResponse response = a2aClient.sendMessage(params);
-
-        return ((Task)response.getResult()).getArtifacts().stream()
-                .flatMap(a -> a.parts().stream())
-                .filter(TextPart.class::isInstance)
-                .map(TextPart.class::cast)
-                .map(TextPart::getText)
-                .collect(Collectors.joining("\n"));
+        Message message =
+                new Message.Builder().role(Message.Role.USER).parts(parts).build();
+        AtomicReference<String> receivedEvent = new AtomicReference<>();
+        AtomicBoolean wasUnexpectedEvent = new AtomicBoolean(false);
+        List<BiConsumer<ClientEvent, AgentCard>> consumers = List.of((event, agentCard) -> {
+            if (event instanceof MessageEvent messageEvent) {
+                receivedEvent.set(messageEvent.getMessage().getParts().stream()
+                        .filter(TextPart.class::isInstance)
+                        .map(TextPart.class::cast)
+                        .map(TextPart::getText)
+                        .collect(Collectors.joining("\n")));
+            } else if (event instanceof TaskEvent taskEvent) {
+                receivedEvent.set(taskEvent.getTask().getArtifacts().stream()
```

Comment:
If the client cannot send a streaming message, then the server should always send back a Message as opposed to a Task and then update events but it's ok to have this handling here.